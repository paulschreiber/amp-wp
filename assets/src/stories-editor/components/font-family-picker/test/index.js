/**
 * External dependencies
 */
import { render } from 'enzyme';

/**
 * Internal dependencies
 */
import FontFamilyPicker from '../';

const { ampStoriesFonts } = window;

describe( 'FontFamilyPicker', () => {
	it( 'should render a default button if no font is selected', () => {
		const fontFamilyPicker = render( <FontFamilyPicker fonts={ ampStoriesFonts } /> );
		expect( fontFamilyPicker ).toMatchSnapshot();
	} );

	it( 'should render the selected font name', () => {
		const fontFamilyPicker = render( <FontFamilyPicker fonts={ ampStoriesFonts } value="Arial" /> );
		expect( fontFamilyPicker ).toMatchSnapshot();
	} );

	it( 'should render the selected font svg preview', () => {
		const fontFamilyPicker = render( <FontFamilyPicker fonts={ ampStoriesFonts } value="Roboto" /> );
		expect( fontFamilyPicker ).toMatchSnapshot();
	} );
} );
