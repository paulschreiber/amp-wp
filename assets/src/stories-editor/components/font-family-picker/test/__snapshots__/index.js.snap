// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FontFamilyPicker should render a default button if no font is selected 1`] = `
<div
  class="components-base-control"
>
  <div
    class="components-base-control__field"
  >
    <label
      class="components-base-control__label"
      for="amp-stories-font-family-picker"
    >
      Font Family
    </label>
    <div
      class="components-preview-picker__buttons"
    >
      <div
        class="components-preview-picker__dropdown"
      >
        <button
          aria-expanded="false"
          aria-label="Font Family: None"
          class="components-button components-preview-picker__selector is-button is-default is-large"
          type="button"
        >
          None
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`FontFamilyPicker should render the selected font name 1`] = `
<div
  class="components-base-control"
>
  <div
    class="components-base-control__field"
  >
    <label
      class="components-base-control__label"
      for="amp-stories-font-family-picker"
    >
      Font Family
    </label>
    <div
      class="components-preview-picker__buttons"
    >
      <div
        class="components-preview-picker__dropdown"
      >
        <button
          aria-expanded="false"
          aria-label="Font Family: Arial"
          class="components-button components-preview-picker__selector is-button is-default is-large"
          type="button"
        >
          Arial
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`FontFamilyPicker should render the selected font svg preview 1`] = `
<div
  class="components-base-control"
>
  <div
    class="components-base-control__field"
  >
    <label
      class="components-base-control__label"
      for="amp-stories-font-family-picker"
    >
      Font Family
    </label>
    <div
      class="components-preview-picker__buttons"
    >
      <div
        class="components-preview-picker__dropdown"
      >
        <button
          aria-expanded="false"
          aria-label="Font Family: Roboto"
          class="components-button components-preview-picker__selector is-button is-default is-large"
          type="button"
        >
          <svg
            height="13"
          >
            <title>
              Roboto
            </title>
            <path
              d="M6.929 11.25L4.094 6.687H.396v4.563H0V.586h3.398a5.034 5.034 0 0 1 1.063.106q.688.148 1.21.505a2.998 2.998 0 0 1 .28.217 2.719 2.719 0 0 1 .923 1.798 3.855 3.855 0 0 1 .025.45 2.898 2.898 0 0 1-.677 1.908q-.678.817-1.725 1.014l2.849 4.563v.103h-.417zm17.029-4.014v.125a6.975 6.975 0 0 1-.095 1.184q-.18 1.044-.704 1.76a2.638 2.638 0 0 1-.909.801q-.513.268-1.156.289a3.389 3.389 0 0 1-.11.001 3.697 3.697 0 0 1-.92-.11 3.1 3.1 0 0 1-.677-.259 2.792 2.792 0 0 1-1.106-1.029v1.252h-.395V0h.395v4.768a3.223 3.223 0 0 1 .519-.725 2.722 2.722 0 0 1 .584-.461 2.916 2.916 0 0 1 1.142-.377 3.694 3.694 0 0 1 .443-.026 2.912 2.912 0 0 1 1.033.176 2.557 2.557 0 0 1 1.157.889q.632.844.764 2.228a8.084 8.084 0 0 1 .035.764zm10.891-6.049h.402v2.138h1.802v.403h-1.802v5.625a4.122 4.122 0 0 0 .022.439q.049.45.204.727a1.067 1.067 0 0 0 .049.079q.255.374.83.401a1.941 1.941 0 0 0 .089.002 3.983 3.983 0 0 0 .262-.009q.142-.01.305-.029a8.473 8.473 0 0 0 .217-.028l.066.374q-.242.087-.828.087a2.454 2.454 0 0 1-.447-.037q-.249-.047-.44-.149a1.092 1.092 0 0 1-.351-.293 1.482 1.482 0 0 1-.221-.416q-.139-.392-.157-.967a5.475 5.475 0 0 1-.002-.13V3.728h-1.465v-.403h1.465V1.187zM9.045 7.485v-.351a5.08 5.08 0 0 1 .134-1.189 4.229 4.229 0 0 1 .302-.851 3.475 3.475 0 0 1 .737-1.025 3.246 3.246 0 0 1 .494-.385 3.252 3.252 0 0 1 1.673-.503 3.909 3.909 0 0 1 .125-.002 3.463 3.463 0 0 1 1.192.201 3.2 3.2 0 0 1 .595.289 3.268 3.268 0 0 1 1.146 1.223 3.967 3.967 0 0 1 .088.169 4.398 4.398 0 0 1 .419 1.478 5.418 5.418 0 0 1 .031.551v.351a5.179 5.179 0 0 1-.122 1.146 4.225 4.225 0 0 1-.31.898 3.434 3.434 0 0 1-.765 1.052 3.248 3.248 0 0 1-.461.354 3.252 3.252 0 0 1-1.674.504 3.909 3.909 0 0 1-.125.001 3.457 3.457 0 0 1-1.2-.204 3.197 3.197 0 0 1-.598-.294 3.309 3.309 0 0 1-1.177-1.284 3.962 3.962 0 0 1-.057-.111 4.393 4.393 0 0 1-.416-1.471 5.398 5.398 0 0 1-.031-.547zm16.465 0v-.351a5.08 5.08 0 0 1 .134-1.189 4.229 4.229 0 0 1 .302-.851 3.475 3.475 0 0 1 .737-1.025 3.246 3.246 0 0 1 .494-.385 3.252 3.252 0 0 1 1.673-.503 3.909 3.909 0 0 1 .125-.002 3.463 3.463 0 0 1 1.192.201 3.2 3.2 0 0 1 .595.289 3.268 3.268 0 0 1 1.146 1.223 3.967 3.967 0 0 1 .088.169 4.398 4.398 0 0 1 .419 1.478 5.418 5.418 0 0 1 .031.551v.351a5.179 5.179 0 0 1-.122 1.146 4.225 4.225 0 0 1-.31.898 3.434 3.434 0 0 1-.765 1.052 3.248 3.248 0 0 1-.462.354 3.252 3.252 0 0 1-1.673.504 3.909 3.909 0 0 1-.125.001 3.457 3.457 0 0 1-1.2-.204 3.197 3.197 0 0 1-.598-.294 3.309 3.309 0 0 1-1.178-1.284 3.962 3.962 0 0 1-.056-.111 4.393 4.393 0 0 1-.416-1.471 5.398 5.398 0 0 1-.031-.547zm12.854 0v-.351a5.08 5.08 0 0 1 .134-1.189 4.229 4.229 0 0 1 .302-.851 3.475 3.475 0 0 1 .737-1.025 3.246 3.246 0 0 1 .494-.385 3.252 3.252 0 0 1 1.673-.503 3.909 3.909 0 0 1 .125-.002 3.463 3.463 0 0 1 1.192.201 3.2 3.2 0 0 1 .595.289 3.268 3.268 0 0 1 1.146 1.223 3.967 3.967 0 0 1 .088.169 4.398 4.398 0 0 1 .419 1.478 5.418 5.418 0 0 1 .031.551v.351a5.179 5.179 0 0 1-.122 1.146 4.225 4.225 0 0 1-.31.898 3.434 3.434 0 0 1-.765 1.052 3.248 3.248 0 0 1-.462.354 3.252 3.252 0 0 1-1.673.504 3.909 3.909 0 0 1-.125.001 3.457 3.457 0 0 1-1.2-.204 3.197 3.197 0 0 1-.598-.294 3.309 3.309 0 0 1-1.178-1.284 3.962 3.962 0 0 1-.056-.111 4.393 4.393 0 0 1-.416-1.471 5.398 5.398 0 0 1-.031-.547zM9.441 7.207v.234a4.34 4.34 0 0 0 .181 1.268 3.919 3.919 0 0 0 .211.545q.392.824 1.091 1.286a2.802 2.802 0 0 0 1.433.457 3.412 3.412 0 0 0 .167.004 2.929 2.929 0 0 0 1.16-.224 2.788 2.788 0 0 0 1.049-.794 3.57 3.57 0 0 0 .757-1.611 5.064 5.064 0 0 0 .096-1.011v-.227q0-.974-.396-1.802a3.339 3.339 0 0 0-.572-.85 2.956 2.956 0 0 0-.523-.443 2.793 2.793 0 0 0-1.558-.465 3.359 3.359 0 0 0-.027 0 2.886 2.886 0 0 0-1.196.243 2.816 2.816 0 0 0-1.016.79 3.67 3.67 0 0 0-.793 1.783 5.02 5.02 0 0 0-.064.817zm16.465 0v.234a4.34 4.34 0 0 0 .181 1.268 3.919 3.919 0 0 0 .211.545q.391.824 1.091 1.286a2.802 2.802 0 0 0 1.433.457 3.412 3.412 0 0 0 .167.004 2.929 2.929 0 0 0 1.16-.224 2.788 2.788 0 0 0 1.049-.794 3.57 3.57 0 0 0 .756-1.611 5.064 5.064 0 0 0 .097-1.011v-.227q0-.974-.396-1.802a3.339 3.339 0 0 0-.572-.85 2.956 2.956 0 0 0-.523-.443 2.793 2.793 0 0 0-1.558-.465 3.359 3.359 0 0 0-.027 0 2.886 2.886 0 0 0-1.196.243 2.816 2.816 0 0 0-1.016.79 3.67 3.67 0 0 0-.793 1.783 5.02 5.02 0 0 0-.064.817zm12.854 0v.234a4.34 4.34 0 0 0 .181 1.268 3.919 3.919 0 0 0 .211.545q.391.824 1.091 1.286a2.802 2.802 0 0 0 1.433.457 3.412 3.412 0 0 0 .167.004 2.929 2.929 0 0 0 1.16-.224 2.788 2.788 0 0 0 1.049-.794 3.57 3.57 0 0 0 .756-1.611 5.064 5.064 0 0 0 .097-1.011v-.227q0-.974-.396-1.802a3.339 3.339 0 0 0-.572-.85 2.956 2.956 0 0 0-.523-.443 2.793 2.793 0 0 0-1.558-.465 3.359 3.359 0 0 0-.027 0 2.886 2.886 0 0 0-1.196.243 2.816 2.816 0 0 0-1.016.79 3.67 3.67 0 0 0-.793 1.783 5.02 5.02 0 0 0-.064.817zM18.281 5.61v3.633a2.792 2.792 0 0 0 .458.806 2.485 2.485 0 0 0 .549.494 2.688 2.688 0 0 0 .962.388 3.744 3.744 0 0 0 .741.07 2.443 2.443 0 0 0 .871-.149q.613-.232 1.019-.825a3.295 3.295 0 0 0 .419-.894q.131-.426.193-.943a8.326 8.326 0 0 0 .054-.983 7.006 7.006 0 0 0-.066-.994q-.145-1.013-.611-1.676-.678-.963-1.893-.963a3.216 3.216 0 0 0-.827.102 2.516 2.516 0 0 0-.862.411 2.561 2.561 0 0 0-.703.81 3.643 3.643 0 0 0-.304.713zM.396.981v5.311h3.361a3.3 3.3 0 0 0 .89-.114 2.494 2.494 0 0 0 1.099-.634 2.512 2.512 0 0 0 .756-1.776 3.321 3.321 0 0 0 .002-.106 2.979 2.979 0 0 0-.118-.86 2.292 2.292 0 0 0-.721-1.103 2.825 2.825 0 0 0-1.1-.57q-.417-.115-.912-.14a5.558 5.558 0 0 0-.284-.008H.396z"
              fill="none"
              fill-rule="evenodd"
              font-size="12"
              stroke="#000"
              stroke-linecap="round"
              stroke-width=".25mm"
              vector-effect="non-scaling-stroke"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>
</div>
`;
